packages/react-query/src/internals/getQueryKey.ts

```diff
@@ -1,11 +1,36 @@
+export type QueryType = 'query' | 'infinite' | 'any';
+
+export type QueryKey = [
+  string[],
+  { input?: unknown; type?: Exclude<QueryType, 'any'> }?,
+];
+
 /**
- * We treat `undefined` as an input the same as omitting an `input`
- * https://github.com/trpc/trpc/issues/2290
- */
+ * To allow easy interactions with groups of related queries, such as
+ * invalidating all queries of a router, we use an array as the path when
+ * storing in tanstack query.
+ **/
 export function getQueryKey(
-  path: string,
+  path: string[],
   input: unknown,
-): [string] | [string, unknown] {
-  if (path.length) return input === undefined ? [path] : [path, input];
-  return [] as unknown as [string];
+  type: QueryType,
+): QueryKey {
+  // Construct a query key that is easy to destructure and flexible for
+  // partial selecting etc.
+  // https://github.com/trpc/trpc/issues/3128
+
+  // some parts of the path may be dot-separated, split them up
+  const splitPath = path.flatMap((part) => part.split('.'));
```

Comment:
is there a reason to support this syntax? does anyone do it, it makes it a hassle to call on the client `trpc.foo["bar.baz"].useQuery`

Reply:
I agree!!

I did put in ab issue that went stale where I thibk the best option was stopping this being allowed at router deffinition level!

#2759

I had it as one of the alternative options but having thought through more I reckon removing capability is the best choice IMHO!